### ISSUE TITLE
S1185: Derived records need to override ToString to prevent default code generation by the compiler

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#5686]
### Description

Reported in the community forum: https://community.sonarsource.com/t/overriding-tostring-for-record-type-can-lead-to-incorrect-code-smell/63372

### Repro steps

```cs
using System;

Console.WriteLine($"ToString without override: {new WithoutOverride()}");
Console.WriteLine($"ToString with override: {new WithOverride()}");

public record Base {
    public override string ToString() => "Expected output";
}

public record WithoutOverride: Base {
}

public record WithOverride: Base {
     public override string ToString() => base.ToString(); // FP: S1185
}
```
Output:
```
WithoutOverride: WithoutOverride { }
WithOverride: Expected output
```

[Sharplab.io](https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYCMBYAUH1TATgAoASAIgBUB7AZWDAEsoBzAAgHcngALGiMHY0AbgFMwzOGJDsA3lDGd2AdR79BAeXGSm0kgEoAvhQMBuAsXLV6jFh259hOqTPmLlavtomvDJ8zwCAGZ2MDEAYxowOHYAIQBDAGcxeTx2DPZUUNFfPVTCAAZ2WgZmNkN2AF4APnYKAFEADwAHSOAxWIFgFsEKC1wjINxssMjo2K8NYB9daVlElLTB4dHwqJjVdVnXBeTUuXTMrJyXfKzMYtK7CoNquoAjfYA6a/LWQwGh/FwgA=)

### Expected behavior

FP. Don't raise S1185

### Actual behavior

S1185 is raised

### Known workarounds

Standard suppress mechanisms

### Related information

* C#/VB.NET Plugins version: Latest
